### ISSUE TITLE
Speaker Change Detection Model

### ISSUE THREAD

**Issue by Alejandro97JS** [pyannote/pyannote-audio#673]
Hello. I am trying to train a SCD model with my own data.

If I follow the tutorial, then I get correct results. But when I train with my own data, I get a low value of the loss function, and then a good percentage of success in the validation step. However, when I test my files, the model does not detect any speaker change, all the speech is associated with one speaker. Furthermore, if I plot scd_scores, the graph is totally flat, there are absolutely no peaks.

I use the same configuration file as the tutorial, although my dataset is smaller. May "per_epoch=1" be a problem for me? My audios have a sample rate of 44.1 kHz, but in feature extraction I use Raw Audio and 16 kHz. UEM files show that all in my audio files is speech, and my RTTM files are properly annotated with several speakers.

What could be going wrong? Any idea? Thank you very much.

**Comment by hbredin** [comment]
It could be because your training files do not contain enough/any speaker change.

How many speakers are there per audio file on average? 
How many speaker changes are there per audio file on average?

Anyway, I'd sugggest you have a look at this [new segmentation model](https://huggingface.co/pyannote/segmentation), part of upcoming pyannote.audio 2.0 (which most likely won't provide a dedicated SCD model any more).

**Comment by Wulin824** [comment]
> It could be because your training files do not contain enough/any speaker change.
> 
> 
> 
> How many speakers are there per audio file on average? 
> 
> How many speaker changes are there per audio file on average?
> 
> 
> 
> Anyway, I'd sugggest you have a look at this [new segmentation model](https://huggingface.co/pyannote/segmentation), part of upcoming pyannote.audio 2.0 (which most likely won't provide a dedicated SCD model any more).

I have the same problem. My training set has 100 hours, and each audio contains only two people's Chinese conversations (similar to one sentence from you to me). What do you mean by including enough speaker change? Please explain, is my data set not good? Thank you!
